test/OrchardCore.Commerce.Tests.UI/Tests/InventoryTests/InventoryBehaviourTests.cs

```diff
@@ -0,0 +1,122 @@
+                    shouldExist: true);
+
+                await context.GoToAdminRelativeUrlAsync($"/Contents/ContentItems/{TestProduct}/Edit");
+                await context.ClickReliablyOnAsync(By.XPath("//input[@id='InventoryPart_AllowsBackOrder_Value']"));
+                await context.ClickAndFillInWithRetriesAsync(
+                    By.XPath("//input[@id='InventoryPart_MinimumOrderQuantity_Value']"), "3");
+                await context.ClickAndFillInWithRetriesAsync(
+                    By.XPath("//input[@id='InventoryPart_MaximumOrderQuantity_Value']"), "10");
+                await context.ClickPublishAsync();
+                await context.GoToRelativeUrlAsync("/cart");
+
+                // When quantity is set to a value below the minimum possible quantity, error message should appear.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "2",
+                    "The checkout quantity for Test Product is less than the minimum allowed (3).",
+                    shouldExist: true);
+
+                // When quantity is set to a value above the minimum possible quantity, error message should not appear.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "4",
+                    "The checkout quantity for Test Product is less than the minimum allowed (3).",
+                    shouldExist: false);
+
+                // When quantity is set to a value above the current inventory value, error message should appear.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "10",
+                    "There are not enough Test Product left in stock.",
+                    shouldExist: true);
+
+                await context.GoToAdminRelativeUrlAsync($"/Contents/ContentItems/{TestProduct}/Edit");
+                await context.ClickReliablyOnAsync(By.XPath("//input[@id='InventoryPart_IgnoreInventory_Value']"));
+                await context.ClickPublishAsync();
+                await context.GoToRelativeUrlAsync("/cart");
+
+                // When inventory is ignored, inventory checks should not apply.
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "2",
+                    "The checkout quantity for Test Product is less than the minimum allowed (3).",
+                    shouldExist: false);
+
+                await UpdateCartAndAssertErrorsAsync(
+                    context,
+                    "15",
+                    "The checkout quantity for Test Product is more than the maximum allowed (10).",
+                    shouldExist: false);
+                AssertError(context, "There are not enough Test Product left in stock.", shouldExist: false);
+            },
+            browser);
+
+    private static async Task UpdateCartAndAssertErrorsAsync(
+        UITestContext context,
+        string quantity,
```

Comment:
Why is `quantity` not `int`?

Reply:
`ClickAndFillInWithRetriesAsync()` needs a `string` for text, figured it's simpler this way than converting.
![image](https://github.com/OrchardCMS/OrchardCore.Commerce/assets/51411356/75e68271-869a-4f55-9b24-ba56700419b9)

Reply:
It's not that complicated to convert and it's worse for both aesthetics and readability to pass pure numbers as string. Also it feels inconsistent because below that method you have QuantityFieldBy where the parameter is int.